### PR TITLE
[KServe] test the missing env var fix

### PR DESCRIPTION
**Pull request by kpouget**

### CODE REVIEW

File: projects/kserve/testing/test_e2e.py [hunk 1]
```diff
@@ -324,10 +324,15 @@ def deploy_consolidated_model(consolidated_model, namespace=None, mute_logs=None
         logging.warning("No secret env key defined for this model")
 
 
-    if (extra_env := consolidated_model["serving_runtime"].get("extra_env")):
+    if (extra_env := consolidated_model["serving_runtime"]["transformer"].get("extra_env")):
```
Comment (dagrayvid) [inline projects/kserve/testing/test_e2e.py:327]: A nit from your past self ;) 

```suggestion
    if (extra_env := consolidated_model["serving_runtime"].get("transformer", {}).get("extra_env")):
```
If the transformer or kserve dicts are missing we will get a keyerror.

Comment (kpouget) [reply projects/kserve/testing/test_e2e.py]: 😀

File: projects/kserve/testing/test_e2e.py [hunk 2]
```diff
@@ -324,10 +324,15 @@ def deploy_consolidated_model(consolidated_model, namespace=None, mute_logs=None
         logging.warning("No secret env key defined for this model")
 
 
-    if (extra_env := consolidated_model["serving_runtime"].get("extra_env")):
+    if (extra_env := consolidated_model["serving_runtime"].get("transformer", {}).get("extra_env")):
```
[the remark this answers is not in the document]
Comment (kpouget) [reply projects/kserve/testing/test_e2e.py:327]: I'm not sure to understand, `extra_env_kserve` `extra_env_transformer` don't exist (`grep`) in the repo 🤔

[the remark this answers is not in the document]
Comment (dagrayvid) [reply projects/kserve/testing/test_e2e.py:327]: Sorry, I was confused! The important thing is that `args_dict["sr_kserve_extra_env_values"]` is set (which you have) not the temporary variable name.

**Comment by kpouget** [comment]
/test rhoai-plot kserve https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_topsail/153/pull-ci-openshift-psap-topsail-main-rhoai-light/1734290349242716160/artifacts/light/test/artifacts/000__local_ci__run_multi_e2e_perf_test/artifacts/ci-pod-0/
/var matbench.workload=projects.kserve.visualizations.kserve-prom

**Comment by kpouget** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_topsail/153/pull-ci-openshift-psap-topsail-main-rhoai-light/1734290349242716160/artifacts/light/test/artifacts/report_index.html)
[visualization replot passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_topsail/153/pull-ci-openshift-psap-topsail-main-rhoai-plot/1734493396661178368/artifacts/plot/test/artifacts/report_04_report:%20LTS.html)

merging :)

**Pull request merged**